miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/file/MiniAppFilePicker.kt

```diff
@@ -0,0 +1,51 @@
+package com.rakuten.tech.mobile.miniapp.file
+
+import android.app.Activity
+import android.content.Context
+import android.net.Uri
+import android.util.Log
+import android.webkit.ValueCallback
+import android.webkit.WebChromeClient
+
+/**
+ * The file picker of a miniapp which needs to be implemented by the HostApp.
+ * @param requestCode of file picking using an intent inside sdk, which will also be used
+ * to retrieve the data by [Activity.onActivityResult] in the HostApp.
+ **/
+open class MiniAppFilePicker(var requestCode: Int) {
+
+    internal var callback: ValueCallback<Array<Uri>>? = null
+
+    /**
+     * HostApp can override [WebChromeClient.onShowFileChooser] to choose a file
+     * which is requested using HTML forms with 'file' input type.
+     * @param filePathCallback The callback to receive the Uris.
+     * @param fileChooserParams The file chooser parameters can be used to create an intent to choose file.
+     * @param context Activity's context from the HostApp.
+     */
+    @Suppress("TooGenericExceptionCaught", "SwallowedException")
+    fun onShowFileChooser(
```

Comment:
I just noticed: it seems a little strange that this class is named `MiniAppFilePicker` but this method is named `onShowFileChooser`. So maybe we should either change the class name to `MiniAppFileChooser` OR change this method name to `onShowFilePicker`. What do you think?

Reply:
Right, so I have renamed it to `MiniAppFileChooser`.